### ISSUE TITLE
sphtfunc.blm_gauss() is not consistent with the rest of healpy

### ISSUE THREAD

**Issue by mreineck** [healpy/healpy#861]
This is a heads-up that the approximations used to compute the Gaussian beam `a_lm` created by `blm_gauss()` are not consistent with the window functions produced by `gauss_beam()`. I'll try to address this in a few weeks' time, but wanted to draw attention to this already now.
The code in `gauss_beam` follows the Challinor et al. paper, https://arxiv.org/pdf/astro-ph/0008228.pdf, while `blm_gauss` is based on the Fortran "Beam" package from Imperial College (?) that was used inside Planck LevelS.

I would greatly appreciate any comments on this! @hivon, which of the alternatives do you suggest?

**Comment by hivon** [comment]
Hi Martin, 
the difference is between l^2 and l*(l+1), in the exp(), is that right ? I have always used the latter in IDL and F90, following Challinor et al.
I thought the LevelS beam package was (co)developped by M. Ashdown, who is a co-author of Challinor et al

**Comment by mreineck** [comment]
> the difference is between l^2 and l*(l+1), in the exp(), is that right ?

Yes, that's the most significant difference, I'd say. (There is also a constant factor I don't understand for the E and B components, but that doesn't feel so worrying.)

> I thought the LevelS beam package was (co)developped by M. Ashdown, who is a co-author of Challinor et al

Correct, and this makes things even more confusing to me. If you look at, e.g., https://gitlab.mpcdf.mpg.de/mtr/levels/-/blame/master/Beam/beam_alm.f90#L150, it's obvious that the `l^2` formula is used there. The source code even mentions the Challinor et al. paper...

Unfortunately I no longer have the LevelS CVS history of the file in question, so I can't do any archeology that might have given us more hints.

P.S.: I just noticed that Mark Ashdown is on Github ... @majashdown, could you perhaps shed some light on this?
